File: SKMiniBank-Android/app/src/main/java/com/bank/skminibank/activities/SplashActivity.java
package com.bank.skminibank.activities;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.view.animation.AccelerateDecelerateInterpolator;
import android.view.View;
import android.widget.ProgressBar;

import androidx.appcompat.app.AppCompatActivity;

import com.bank.skminibank.R;
import com.bank.skminibank.utils.SessionManager;

public class SplashActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);

        View card = findViewById(R.id.containerMain);
        ProgressBar progressBar = findViewById(R.id.splashProgress);

        // 1. Premium Entrance Animation
        card.setAlpha(0f);
        card.setScaleX(0.9f);
        card.setScaleY(0.9f);
        
        card.animate()
                .alpha(1f)
                .scaleX(1f)
                .scaleY(1f)
                .setDuration(1200)
                .setInterpolator(new AccelerateDecelerateInterpolator())
                .start();

        // 2. Progress Bar Logic
        android.animation.ObjectAnimator progressAnimator = android.animation.ObjectAnimator.ofInt(progressBar, "progress", 0, 100);
        progressAnimator.setDuration(2500);
        progressAnimator.setInterpolator(new android.view.animation.LinearInterpolator());
        progressAnimator.start();

        // 3. Transition to Login or Dashboard
        new Handler().postDelayed(() -> {
            if (!isFinishing()) {
                SessionManager sessionManager = new SessionManager(SplashActivity.this);
                Intent intent;
                if (sessionManager.isLoggedIn()) {
                    intent = new Intent(SplashActivity.this, DashboardActivity.class);
                } else {
                    intent = new Intent(SplashActivity.this, LoginActivity.class);
                }
                startActivity(intent);
                overridePendingTransition(android.R.anim.fade_in, android.R.anim.fade_out);
                finish();
            }
        }, 3000);
    }
}

File: android-user-app/app/src/main/java/com/bank/userapp/SplashActivity.kt
package com.bank.userapp

import android.content.Intent
import android.graphics.drawable.AnimationDrawable
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import android.view.animation.AnimationUtils
import android.widget.ImageView
import android.widget.ProgressBar
import android.widget.RelativeLayout
import androidx.appcompat.app.AppCompatActivity

class SplashActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)

        val layout = findViewById<RelativeLayout>(R.id.splash_layout)
        val animationDrawable = layout.background as AnimationDrawable
        animationDrawable.setEnterFadeDuration(2000)
        animationDrawable.setExitFadeDuration(2000)
        animationDrawable.start()

        val logoImg = findViewById<ImageView>(R.id.logo_img)
        val rotateAnimation = AnimationUtils.loadAnimation(this, R.anim.rotate)
        logoImg.startAnimation(rotateAnimation)

        val progressBar = findViewById<ProgressBar>(R.id.progress_bar)
        
        val handler = Handler(Looper.getMainLooper())
        var progressStatus = 0
        
        Handler(Looper.getMainLooper()).postDelayed({
            val session = UserSession(this)
            if (session.isLoggedIn()) {
                startActivity(Intent(this, MainActivity::class.java))
            } else {
                startActivity(Intent(this, LoginActivity::class.java))
            }
            finish()
        }, 3000)
    }
}
